devlib/utils/types.py

```diff
@@ -68,6 +68,15 @@ def numeric(value):
     """
     if isinstance(value, int):
         return value
+
+    if isinstance(value, basestring):
+        value = value.strip()
+        if value.endswith('%'):
+            try:
+                return float(value.rstrip('%')) / 100
```

Comment:
No worries -- it's good to have the discussion. What you are suggesting are perfectly valid semantics that are useful in certain situations. However,  extending `numeric` with them would make it unsuitable for its intended purpose. For what you're describing, WA has the `list_of` and `list_or` families of functions, e.g. https://github.com/ARM-software/workload-automation/blob/master/wlauto/utils/types.py#L111

(These have not been migrated into devlib because, as of now, there is no clear need for them in the code line; but they could be, if and when such a need arises.)